### PR TITLE
handle RSE not found in getWritePfn fix #7107 fix #8230

### LINKED ISSUE #7107: 'didDict' referenced before assignment in TW stageout check
serendiptious finding in tw log
```
2022-03-04 05:41:48,470:INFO:StageoutCheck:Will check stageout at T2_CH_CERNBOX
2022-03-04 05:41:48,470:WARNING:RucioUtils:Try Rucio lfn2pn with operation third_party_copy
2022-03-04 05:41:48,498:WARNING:RucioUtils:Rucio lfn2pfn resolution for third_party_copy failed with:
RSE does not exist.
Details: RSE 'T2_CH_CERNBOX' cannot be found in vo 'def'
Try next one.
2022-03-04 05:41:48,498:WARNING:RucioUtils:Try Rucio lfn2pn with operation write
2022-03-04 05:41:48,508:WARNING:RucioUtils:Rucio lfn2pfn resolution for write failed with:
RSE does not exist.
Details: RSE 'T2_CH_CERNBOX' cannot be found in vo 'def'
Try next one.
2022-03-04 05:41:48,508:WARNING:RucioUtils:Try Rucio lfn2pn with operation read
2022-03-04 05:41:48,528:WARNING:RucioUtils:Rucio lfn2pfn resolution for read failed with:
RSE does not exist.
Details: RSE 'T2_CH_CERNBOX' cannot be found in vo 'def'
Try next one.
2022-03-04 05:41:48,529:INFO:ServerUtilities:Warning: Failed to delete file 220304044113hwencrabskim6900to4muxining_crab3check.tmp
2022-03-04 05:41:48,532:ERROR:Handler:Problem handling 220304_044113:hwen_crab_skim_X6900to4mu_xining because of local variable 'didDict' referenced before assignment failure, traceback follows
Traceback (most recent call last):
  File "/data/srv/TaskManager/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/slc7_amd64_gcc630/cms/crabtaskworker/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/lib/python3.8/site-packages/TaskWorker/Actions/Handler.py", line 80, in executeAction
    output = work.execute(nextinput, task=self._task, tempDir=self.tempDir)
  File "/data/srv/TaskManager/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/slc7_amd64_gcc630/cms/crabtaskworker/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/lib/python3.8/site-packages/TaskWorker/Actions/StageoutCheck.py", line 96, in execute
    pfn = getWritePFN(self.rucioClient, siteName=self.task['tm_asyncdest'], lfn=lfn, logger=self.logger)
  File "/data/srv/TaskManager/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/slc7_amd64_gcc630/cms/crabtaskworker/py3.220225-c05c3bac4cb7fda7b0c4ade9224b6414/lib/python3.8/site-packages/RucioUtils.py", line 56, in getWritePFN
    if not didDict:
UnboundLocalError: local variable 'didDict' referenced before assignment

```

### LINKED ISSUE #8230: tasks stuck in QUEUED if Rucio quota check fails
CRAB project directory:		/afs/cern.ch/work/b/belforte/CRAB3/TC3/crab_20240205_090454
Task name:			240205_080458:belforte_crab_20240205_090454
Grid scheduler - Task Worker:	crab3@vocms059.cern.ch - crab-dev-tw01
Status on the CRAB server:	QUEUED on command SUBMIT
Task URL to use for HELP:	https://cmsweb-test2.cern.ch/crabserver/ui/task/240205_080458%3Abelforte_crab_20240205_090454
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=belforte&var-task=240205_080458%3Abelforte_crab_20240205_090454&from=1707116699000&to=now
Log file is /afs/cern.ch/work/b/belforte/CRAB3/TC3/crab_20240205_090454/crab.log

in TW log:
```
Starting <TaskWorker.Actions.StageoutCheck.StageoutCheck object at 0x7ff96ff20fd0> on 240205_080458:belforte_crab_20240205_090454
Checking Rucio quota from account belforte.
Problem handling 240205_080458:belforte_crab_20240205_090454 because of RSE does not exist.
Details: RSE 'T2_CH_CERNBOX' cannot be found in vo 'def' failure, traceback follows
Traceback (most recent call last):
  File "/data/repos/CRABServer/src/python/TaskWorker/Actions/Handler.py", line 86, in executeAction
    output = work.execute(nextinput, task=self._task, tempDir=self.tempDir)
  File "/data/repos/CRABServer/src/python/TaskWorker/Actions/StageoutCheck.py", line 87, in execute
    quotaCheck = isEnoughRucioQuota(self.rucioClient, self.task['tm_asyncdest'], rucioAccount)
  File "/data/repos/CRABServer/src/python/ServerUtilities.py", line 1097, in isEnoughRucioQuota
    quotas = list(rucioClient.get_local_account_usage(account, site))
  File "/data/srv/TaskManager/v3.240122-42ae10d09d91cac3852a0ea083265c73/slc7_amd64_gcc630/external/py3-rucio-clients/1.29.10/lib/python3.8/site-packages/rucio/client/accountclient.py", line 327, in get_local_account_usage
    raise exc_cls(exc_msg)
rucio.common.exception.RSENotFound: RSE does not exist.
Details: RSE 'T2_CH_CERNBOX' cannot be found in vo 'def'
```

### PR DESCRIPTION
**Pull request by belforte**

### CODE REVIEW

File: src/python/RucioUtils.py [hunk 1]
```diff
@@ -65,18 +66,22 @@ def getWritePFN(rucioClient=None, siteName='', lfn='',
...
+            msg = f"Site {siteName} not found in CMS site list"
+            raise TaskWorkerException(msg)
         except Exception as ex:
             msg = 'Rucio lfn2pfn resolution for %s failed with:\n%s\nTry next one.'
             logger.warning(msg, operation, str(ex))
             ex_str += "operation: %s, exception: %s\n" % (operation, ex)
-            didDict = None
     if not didDict:
-        msg = 'lfn2pfn resolution with Rucio failed for site: %s  LFN: %s' % (siteName, lfn)
+        msg = ('lfn2pfn eicijijjjiflirdbtgginfbvkcibfvggttujrjhiiitk'
```
Comment (novicecpp) [inline src/python/RucioUtils.py:83]: Yubikey?

Comment (belforte) [reply src/python/RucioUtils.py:83]: yes. oops  just fixed

**Comment by belforte** [comment]
tested on crab-dev-tw01. Now crab status prints
```txt
Rucio client intialized for account belforte
CRAB project directory:		/afs/cern.ch/work/b/belforte/CRAB3/TC3/crab_20240212_173308
Task name:			240212_163314:belforte_crab_20240212_173308
Grid scheduler - Task Worker:	crab3@vocms059.cern.ch - crab-dev-tw01
Status on the CRAB server:	SUBMITFAILED
Task URL to use for HELP:	https://cmsweb-test2.cern.ch/crabserver/ui/task/240212_163314%3Abelforte_crab_20240212_173308
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=belforte&var-task=240212_163314%3Abelforte_crab_20240212_173308&from=1707751994000&to=now
Failure message from server:	Site T2_CH_CERNBOX not found in CMS site list
Log file is /afs/cern.ch/work/b/belforte/CRAB3/TC3/crab_20240212_173308/crab.log
belforte@lxplus806/TC3> 


```

**Comment by belforte** [comment]
rebased and ready for merging

File: src/python/RucioUtils.py [hunk 2]
```diff
@@ -64,26 +67,29 @@ def getWritePFN(rucioClient=None, siteName='', lfn='',
     # "write": provides the PFN to be used with gfal
     # 2022-08: dario checked with felipe that every sane RSE has non-zero value
     # for the third_party_copy_write column, which means that it is available.
-    ex_str = ""
+    exeptionString = ""
```
Comment (novicecpp) [inline src/python/RucioUtils.py:70]: Typo, `exceptionString`.

File: src/python/RucioUtils.py [hunk 3]
```diff
@@ -64,26 +67,29 @@ def getWritePFN(rucioClient=None, siteName='', lfn='',
...
+        except Exception as ex:  # pylint: disable=broad-except
             msg = 'Rucio lfn2pfn resolution for %s failed with:\n%s\nTry next one.'
             logger.warning(msg, operation, str(ex))
-            ex_str += "operation: %s, exception: %s\n" % (operation, ex)
-            didDict = None
+            exeptionString += f"operation: {operation}, exception: {ex}\n"
     if not didDict:
-        msg = 'lfn2pfn resolution with Rucio failed for site: %s  LFN: %s' % (siteName, lfn)
-        msg += ' with exception(s) :\n%s' % str(ex_str)
+        msg = (f"lfn2pfn resolution with Rucio failed for site: {siteName}  LFN: {lfn}")
```
Comment (novicecpp) [inline src/python/RucioUtils.py:85]: Could you remove the parentheses from this line? 
It does not break, but to reduce confusion of the syntax.

**Review by novicecpp — APPROVED** [review]
The fix is LGTM, with some small inline comments.

Thanks Stefano.

File: src/python/RucioUtils.py (hunk 2 above)
Comment (belforte) [reply src/python/RucioUtils.py:70]: OMG  !! thank you !!!

**Pull request merged**